admin-dev/themes/new-theme/js/components/auto-complete-search.js

```diff
@@ -0,0 +1,188 @@
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+/**
+ * This component is used with TypeaheadType forms, and is tightly linked to the content of this
+ * twig file src/PrestaShopBundle/Resources/views/Admin/TwigTemplateForm/typeahead.html.twig
+ *
+ * @todo: the component relies on this TypeaheadType because it was the historic type but it would be worth
+ * creating a new clean form type
+ */
+export default class AutoCompleteSearch {
+  constructor($searchInput, config, dataSetConfig) {
+    this.$searchInput = $searchInput;
+    this.$autoCompleteSearchContainer = this.$searchInput.closest('.autocomplete-search');
+    this.searchInputId = this.$searchInput.prop('id');
+    this.searchInputFullName = this.$autoCompleteSearchContainer.data('fullname');
+    this.$selectionContainer = $(`#${this.searchInputId}-data`);
+
+    // Merge default and input config
+    const inputConfig = config || {};
+    const defaultConfig = {
+      minLength: 2,
+      highlight: true,
+      cache: false,
+      hint: false,
+    };
+    this.config = {...defaultConfig, ...inputConfig};
+
+    // Merge default and input dataSetConfig
+    const inputDataSetConfig = dataSetConfig || {};
+    const defaultDataSetConfig = {
+      display: 'name', // Which field of the object from the list is used for display (can be a string or a callback)
+      value: 'id', // Which field of the object from the list is used for value (can be a string or a callback)
+      limit: 20, // Limit the number of displayed suggestion
+      dataLimit: 0, // How many elements can be selected max
+    };
+    this.dataSetConfig = {...defaultDataSetConfig, ...inputDataSetConfig};
+
+    // Merging object works fine on one level, but on two it erases sub elements even if not present, so
+    // we handle templates separately
+    const defaultTemplates = { // Default rendering functions which can be overridden
+      suggestion(item) {
+        return `<div><img src="${item.image}" style="width:50px" /> ${item.name}</div>`;
+      },
+      pending(query) {
+        return `<div class="px-2">Searching for "${query.query}"</div>`;
+      },
+      notFound(query) {
+        return `<div class="px-2">No results found for "${query.query}"</div>`;
+      },
+      renderSelected: (selectedItem) => `<li>
+            <div>
+              <img src="${selectedItem.image}" style="width:50px" /> ${selectedItem.name}
+            </div>
+          </li>`,
+    };
+    this.dataSetConfig.templates = {...defaultTemplates, ...inputDataSetConfig.templates};
+
+    this.buildTypeahead();
+  }
+
+  /**
+   * Force set values, the selected items are removed Then the component loops through the inputs
+   * and render each object in the list as if it came from the API so the format must match.
+   *
+   * @param values {array}
+   */
+  setValue(values) {
+    this.clearValue();
+    if (!values || values.length <= 0) {
```

Comment:
can it be inferior to 0 ?

Reply:
very unlikely I'm just used to writing it this way for extra care 😅